### PR TITLE
Added Some Unit Tests

### PR DESCRIPTION
**Pull request by vpn1997**
- Added some unit test for views.py

### CODE REVIEW

**Comment by vpn1997** [comment]
ill be adding some more unit test functions. :smile:

File: api/views.py [hunk 1]
```diff
@@ -312,7 +309,7 @@ def custom_demo_controller(request, user_id, id):
             try:
                 demo = Demo.objects.get(id=id, user_id=user_id)
             except Exception:
-                return Response({"text": "Not Found"})
+                 return Response({"text": "Not Found"})
```
Comment (AvaisP) [inline api/views.py]: Remove the extra space

File: requirements.txt [hunk 2]
```diff
@@ -1,31 +1,52 @@
...
 hyperlink==17.2.0
 idna==2.5
 incremental==17.5.0
+itsdangerous==0.24
+Jinja2==2.10
+MarkupSafe==1.0
 msgpack-python==0.4.8
+numpy==1.14.0
 oauthlib==2.0.2
-psycopg2==2.7.1
```
Comment (AvaisP) [inline requirements.txt]: Why are you removing this?

**Comment by AvaisP** [comment]
@vpn1997 Please autoformat the PEP8 errors

File: api/views.py [hunk 3]
```diff
@@ -194,11 +194,6 @@ def custom_component_controller(request, type_req, user_id, demoid):
     elif type_req == "output":
         model = OutputComponent
         serializer = OutputComponentSerializer
-    else:
```
Comment (vpn1997) [inline api/views.py]: type_req can be either input or output as defined in the url pattern .So this code was redundant .

File: django_server/urls.py [hunk 4]
```diff
@@ -24,7 +24,7 @@
 from rest_framework import routers
 
 router = routers.DefaultRouter()
-router.register(r'demo', DemoViewSet, r"demo")
+router.register(r'demo-view', DemoViewSet, r"demo-view")
```
Comment (vpn1997) [inline django_server/urls.py]: api/demo route was already there for custom_demo_controller . So had to change the route for DemoViewSet.

**Comment by vpn1997** [comment]
@AvaisP @uttu357 hey can you please review this pr.

File: api/tests.py [hunk 5]
```diff
@@ -696,3 +843,117 @@ def test_is_cloudcv(self):
...
+        response = self.client.get('/api/output-component/')
+        response = json.loads(response.content.decode('utf-8'))[0]
+        self.assertEqual(response["user_id"], output_component["user_id"])
+        self.assertEqual(response["props"], output_component["props"])
+        self.assertEqual(
+            response["base_component_id"],
+            output_component["base_component_id"])
+
+
+class MyTests(ChannelTestCase):
```
Comment (AvaisP) [inline api/tests.py]: Rename this to ChannelTests

**Comment by vpn1997** [comment]
@AvaisP Done

**Pull request merged**